common/chat-template.hpp

```diff
@@ -0,0 +1,247 @@
+/*
+    Copyright 2024 Google LLC
+
+    Use of this source code is governed by an MIT-style
+    license that can be found in the LICENSE file or at
+    https://opensource.org/licenses/MIT.
+*/
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "minja.hpp"
+#include <json.hpp>
+#include <string>
+#include <vector>
+
+using json = nlohmann::ordered_json;
+
+namespace minja {
+
+class chat_template {
```

Comment:
One idea to be able to `#include "chat-template.hpp"` in main is to forward declare `json` here without `#include <json.hpp>`, only define the prototype of `class chat_template` here. Then we will need a new file `chat-template.cpp` that hold the actual implementation, including `#include <json.hpp>`

Reply:
(Not sure if this even works, but we can do in another PR, just noting my idea here)

Reply:
I was hoping to keep minja header-only for now, but happy to explore options as follow up :-)